desdeo/problem/testproblems.py

```diff
@@ -1572,6 +1574,204 @@ def simple_knapsack_vectors():
+
+    selected_df_p = df.filter(pl.col("holding") == holding).select(["unit", "schedule", "harvest_value_period_2025", "harvest_value_period_2030", "harvest_value_period_2035"])
+    selected_df_p.group_by(["unit", "schedule"])
+    rows_by_key = selected_df_p.rows_by_key(key=["unit", "schedule"])
+    p_array = np.zeros((selected_df_p["unit"].n_unique(), selected_df_p["schedule"].n_unique()))
+    for i in range(np.shape(p_array)[0]):
+        for j in range(np.shape(p_array)[1]):
+            if (unique_units[i], j) in rows_by_key:
+                p_array[i][j] = sum(rows_by_key[(unique_units[i], j)][0])
+
+    constants = []
+    variables = []
+    constraints = []
+    f_1_func = []
+    f_2_func = []
+    f_3_func = []
+    # define the constants V, W and P, decision variable X, constraints, and objective function expressions in one loop
+    for i in range(np.shape(v_array)[0]):
+        # Constants V, W and P
+        v = TensorConstant(
+            name=f"V_{i+1}",
+            symbol=f"V_{i+1}",
+            shape=[np.shape(v_array)[1]], # NOTE: vectors have to be of form [2] instead of [2,1] or [1,2]
+            values=v_array[i].tolist()
+        )
+        constants.append(v)
+        w = TensorConstant(
+            name=f"W_{i+1}",
+            symbol=f"W_{i+1}",
+            shape=[np.shape(w_array)[1]], # NOTE: vectors have to be of form [2] instead of [2,1] or [1,2]
+            values=w_array[i].tolist()
+        )
+        constants.append(w)
+        p = TensorConstant(
+            name=f"P_{i+1}",
+            symbol=f"P_{i+1}",
+            shape=[np.shape(p_array)[1]], # NOTE: vectors have to be of form [2] instead of [2,1] or [1,2]
+            values=p_array[i].tolist()
+        )
+
+        # Decision variable X
+        constants.append(p)
+        x = TensorVariable(
+            name=f"X_{i+1}",
+            symbol=f"X_{i+1}",
+            variable_type=VariableTypeEnum.binary,
+            shape=[np.shape(v_array)[1]], # NOTE: vectors have to be of form [2] instead of [2,1] or [1,2]
+            lowerbounds=np.shape(v_array)[1] * [0],
+            upperbounds=np.shape(v_array)[1] * [1],
+            initial_values=np.shape(v_array)[1] * [0]
+        )
+        variables.append(x)
+
+        # Constraints
+        con = Constraint(
+            name=f"x_con_{i+1}",
+            symbol=f"x_con_{i+1}",
+            cons_type=ConstraintTypeEnum.EQ,
+            func=f"Sum(X_{i+1}) - 1"
+        )
+        constraints.append(con)
+
+        # Objective function expressions
+        exprs = f"V_{i+1}@X_{i+1}"
+        f_1_func.append(exprs)
+
+        exprs = f"W_{i+1}@X_{i+1}"
+        f_2_func.append(exprs)
+
+        exprs = f"P_{i+1}@X_{i+1}"
+        f_3_func.append(exprs)
+
+    # form the objective function sums
+    f_1_func = " + ".join(f_1_func)
+    f_2_func = " + ".join(f_2_func)
+    f_3_func = " + ".join(f_3_func)
+
+    f_1 = Objective(
```

Comment:
They do, but the ideal and nadir values are different for each holding. I suppose the ideal values could be computed for each holding separately but I don't know about available ways to estimate the nadirs.

Reply:
I see. There seems sth we need to do. The current implementation expects explicit nadir and ideal values.

Reply:
As @Matskuu said, each forest holding has their unique ideal and nadir that each of the holding owner of course would be interested. For estimating nadir, you can check out the payoff table method, which was implemented in old desdeo (not sure if in desdeo2 yet).

However, it would be useful to get the ideal and nadir for the whole landscape also. So the combined from all of the holdings. Can we in the current implementation get all holding's solutions and hence the landscape level ideal and nadir or not?

Reply:
Currently the problem is assuming that a holding number is given and only calculates for that holding.